project.py

```diff
@@ -147,19 +168,31 @@ def run_tasks(tasks: Sequence[Task]) -> List[TaskResult]:
   return results
 
 
-def build_tasks(args: argparse.Namespace) -> List[Task]:
-  '''Build a list of tasks based on the parsed arguments.'''
+def create_tasks(args: argparse.Namespace) -> List[Task]:
+  '''Create a list of tasks based on the parsed arguments.'''
   tasks = []
   if args.setup:
-    tasks.append(Task('Set up and ensure dependencies', setup_environment))
+    plan = SetupPlan(
+      install_dependencies=True,
+      required_tools=[Tool('cmake'), Tool('make')],
+      cpp_compilers=['clang++', 'g++', 'clang++18', 'g++14'],
```

Comment:
Instead, I somewhat believe that the functionalities of `compiler_finder.py` should be moved to `automation`. And not hardcoded C++ compilers here - find the compilers in PATH and check them instead.